admin/adminContent.tsx

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect, useContext } from "react";
+import firebase from "../utils/firebase";
+
+import { ContentList } from "./contentList";
+import { AddLink } from "./addLink";
+import { AddField } from "./addField";
+import { AuthContext } from "./authContext";
+
+export const AdminContent = () => {
+  const initialLinks: any = [];
+  const [links, setLinks] = useState(initialLinks);
+
+  const { handleSignOut } = useContext(AuthContext);
+
+  const handleUpdate = async () => {
+    const updatedLinks = await getLinksFromDatabase();
+    setLinks(updatedLinks);
+  };
+
+  useEffect(() => {
+    handleUpdate();
+  }, []);
+
+  return (
+    <>
+      <button onClick={() => handleSignOut()}>Sign out</button>
+      <ContentList handleUpdate={handleUpdate} links={links} />
+      <AddLink handleUpdate={handleUpdate} />
+      <AddField />
+    </>
+  );
+};
+
+const getLinksFromDatabase = async () => {
+  let links: Array<any> = [];
+  try {
+    const snapshot = await firebase
+      .firestore()
+      .collection("resources")
+      .orderBy("title")
+      .get();
+    links = snapshot.docs.map(doc => {
+      const data = doc.data();
+      const id = doc.id;
+      return {
+        title: data.title,
+        url: data.url,
+        id: id,
+        tags: data.tags,
+        date: data.date
+      };
+    });
+  } catch (e) {
+    console.error("Error getting documents: ", e);
+  }
+  return links;
```

Comment:
Istedenfor å ha en egen variabel `links` her, så kan du returnere ting to ganger:

```typescript
const getLinksFromDatabase = async () => {
  try {
    const snapshot = await firebase
      .firestore()
      .collection("resources")
      .orderBy("title")
      .get();
    // returner her!
    return snapshot.docs.map(doc => {
      const data = doc.data();
      const id = doc.id;
      return {
        title: data.title,
        url: data.url,
        id: id,
        tags: data.tags,
        date: data.date
      };
    });
  } catch (e) {
    console.error("Error getting documents: ", e);
    return [];
  }
```

Da synes i alle fall jeg det blir lettere å følge "flyten" i programmet på et vis.